src/SFA.DAS.Commitments.Api.UnitTests/Orchestrators/Mappers/WhenMappingACommitmentView.cs

```diff
@@ -25,20 +21,30 @@ public void Arrange()
             _mapper = new CommitmentMapper(_rules);
         }
 
-        [TestCase(CallerType.Employer)]
-        [TestCase(CallerType.Provider)]
-        public void ThenTransferSenderFieldsAreMappedCorrectly(CallerType callerType)
+        [TestCase(CallerType.Employer, true, false, TransferApprovalStatus.Pending)]
+        [TestCase(CallerType.Provider,false, true, TransferApprovalStatus.Pending)]
+        [TestCase(CallerType.TransferSender, false, false, TransferApprovalStatus.Pending)]
+        public void ThenTransferSenderFieldsAreMappedCorrectly(CallerType callerType, bool canEmployerApprove, bool canProviderApprove, TransferApprovalStatus transferStatus)
         {
             var commitment = new Commitment
             {
                 TransferSenderId = 1,
-                TransferSenderName = "Transfer Sender Org"
+                TransferSenderName = "Transfer Sender Org",
+                TransferApprovalStatus = transferStatus,
+                TransferApprovalActionedOn = new DateTime(2018, 09, 09),
+                TransferApprovalActionedByEmployerName = "Name",
+                ProviderCanApproveCommitment = canProviderApprove,
+                EmployerCanApproveCommitment = canEmployerApprove,
             };
 
             var result = _mapper.MapFrom(commitment, callerType);
 
-            Assert.AreEqual(commitment.TransferSenderId, result.TransferSenderId);
-            Assert.AreEqual(commitment.TransferSenderName, result.TransferSenderName);
+            Assert.AreEqual(commitment.TransferSenderId, result.TransferSenderInfo.TransferSenderId);
```

Comment:
I'm not a fan of using objects passed into the code under test (CUT) as the expected in the asserts, because the CUT could change the fields in the object and the test could pass when it shouldn't. Better would be to have constants that populate the Commitment and are used as the expected in the Asserts, or clone the Commitment and then Assert using the clone's fields.